lib/Core/Executor.cpp

```diff
@@ -4827,53 +4918,88 @@ void Executor::executeMemoryOperation(ExecutionState &state,
       ExecutionState *bound_inner = branches_inner.first;
       ExecutionState *unbound_inner = branches_inner.second;
       if (bound_inner) {
+        ObjectState *wos = bound_inner->addressSpace.getWriteable(mo, os);
```

Comment:
Maybe it's better to call getWriteable only when the type in the ObjectState changes? At least write a comment